packages/uni_app/lib/view/academic_path/exam_page.dart

```diff
@@ -0,0 +1,159 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:uni/controller/local_storage/preferences_controller.dart';
+import 'package:uni/generated/l10n.dart';
+import 'package:uni/model/entities/exam.dart';
+import 'package:uni/model/providers/lazy/exam_provider.dart';
+import 'package:uni/utils/date_time_formatter.dart';
+import 'package:uni/utils/string_formatter.dart';
+import 'package:uni/view/common_widgets/expanded_image_label.dart';
+import 'package:uni/view/lazy_consumer.dart';
+import 'package:uni/view/locale_notifier.dart';
+import 'package:uni_ui/cards/exam_card.dart';
+import 'package:uni_ui/cards/timeline_card.dart';
+
+class ExamsPage extends StatefulWidget {
+  const ExamsPage({super.key});
+
+  @override
+  State<StatefulWidget> createState() => _ExamsPageState();
+}
+
+class _ExamsPageState extends State<ExamsPage> {
+  List<String> hiddenExams = PreferencesController.getHiddenExams();
+  Map<String, bool> filteredExamTypes =
+      PreferencesController.getFilteredExams();
+
+  @override
+  Widget build(BuildContext context) {
+    /*
+      If we want to filters exams again
+        filteredExamTypes[Exam.getExamTypeLong(exam.examType)] ??
+     */
+
+    return LazyConsumer<ExamProvider, List<Exam>>(
+      builder: (context, exams) => ListView(
+        children: _examsByMonth(exams)
+            .entries
+            .map(
+              (entry) => Padding(
+                padding: const EdgeInsets.all(8),
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    Text(
+                      DateTime(
+                        int.parse(entry.key.split('-')[0]),
+                        int.parse(entry.key.split('-')[1]),
+                      )
+                          .fullMonth(
+                            Provider.of<LocaleNotifier>(context).getLocale(),
+                          )
+                          .capitalize(),
+                      style: Theme.of(context).textTheme.headlineMedium,
+                    ),
+                    const SizedBox(height: 8),
+                    ListView.builder(
```

Comment:
Why is there a ListView.builder inside another ListView?

Reply:
which line is the ListView?

Reply:
35, 56

Reply:
humn, I see. What you suggest?